samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -69,16 +69,19 @@
 
   private final StreamGraphBuilder graphBuilder;
 
+  private final ApplicationRunner runner;
+
   private ContextManager contextManager;
 
-  public StreamOperatorTask(StreamGraphBuilder graphBuilder) {
+  public StreamOperatorTask(StreamGraphBuilder graphBuilder, ApplicationRunner runner) {
     this.graphBuilder = graphBuilder;
+    this.runner = runner;
   }
 
   @Override
   public final void init(Config config, TaskContext context) throws Exception {
     // create the MessageStreamsImpl object and initialize app-specific logic DAG within the task
-    StreamGraphImpl streams = new StreamGraphImpl();
+    StreamGraphImpl streams = new StreamGraphImpl(this.runner);
```

Comment:
Unrelated to this RB, but:
1. Don't think this needs to wait for init, should assign in constructor.
2. Prefer renaming to `streamGraph` for clarity.